### DISCUSSION TITLE
“Add to Home Screen” no longer works properly because PWA metadata (ex. `/apple-icon.png`) fails to resolve correctly

### DISCUSSION THREAD

**Discussion by huw** [community/community#27431] [Codespaces] [answered]
I recently deleted my Codespace from my iPad Home Screen, and I wanted to re-add it. Unfortunately, when I re-added it it no longer launched in full screen mode and was lacking a proper home screen icon (which, in turn, means I couldn’t use keyboard shortcuts or trackpad scrolling).

I checked the Web Inspector and noticed that `/apple-icon.png` and `/manifest.json`, among others, no longer resolve correctly. Instead, they show the ordinary Codespace loading screen (which, once it loads, does not load these files either). I’m pretty sure these should resolve to the correct files again.

**Comment by jungaretti** [comment, marked as answer]
If you don't mind sharing, which continent are you using Codespaces from? We suspect this may be a bug in our latest update for the web and will take a look.

**Reply by KartikM123** [reply]
@huw Thanks for sharing this bug! If you don't mind, it would be really helpful if you could also forward the `sessionId` which should be logged in your console. 

Here's a snippet of the sessionId log from my console. 
<img width="259" alt="image" src="https://user-images.githubusercontent.com/19193984/181607165-d0bc6c96-4a83-42db-a745-f52b9894140a.png">

The format of your sessionId will make it easy for us to tell which of our pods is serving your codespace 😄

**Reply by huw** [reply]
I am using Codespaces from Australia (if it's any help, this happens regardless of the region the Codespace is deployed in). Here's my init:

```json
{
  "browserId": "7ed96e7125348612eabbe6eb28f60461",
  "sessionId": "84a985ea-2f29-4236-8662-f46c19cbfa81",
  "pageLoadId": "c2290be1ee5dcea0ed0c21e7e645d9d1",
  "host": "github.dev",
  "browserName": "safari",
  "browserVersion": "15.5.0",
  "browserOS": "Mac OS",
  "vscodeCommit": "3b889b090b5ad5793f524b5d1d39fda662b96a2a",
  "vscodeQuality": "stable",
  "portalName": "codespaces-web-client",
  "version": "0.1.02032.168-6f1b9d0f",
  "stableVSCodeCommitId": "3b889b090b5ad5793f524b5d1d39fda662b96a2a",
  "insiderVSCodeCommitId": "b4d5b4ed69af953901051b1f049851b9dca3563e",
  "serviceLocation": "SoutheastAsia",
  "serviceInstance": "SoutheastAsia",
  "environment": "Production",
  "workbenchType": "codespaces"
}
```

**Reply by huw** [reply]
I worked around the issue by adding the Codespace to my Home Screen on a VPN targeted at a US East location—cheers for the hint. I’ll leave the issue open for now :)

**Reply by jungaretti** [reply]
Can you try again and let us know if this is fixed? If not, could you share the URL of the page that you are using to install the PWA?

**Reply by Magssch** [reply]
I’m also having this problem. I’m unfortunately not able to share my console log (as I’m only travelling with an iPad), but I’m currently located in Western Europe. I tried the fix proposed above (adding the pwa my vpn on my iPad with a vpn targeted at us east) but that workaround is not working for me unfortunately. If it’s any use I’m using the URL https://magssch-tdt4100-exam-2022-jgppgwx62599v.github.dev/ (please note it’s a private repo). Changing the codespaces server location does not seem to work either (as noted above)

**Reply by huw** [reply]
@jungaretti No dice—`/apple-icon.png` and `/manifest.json` still broken. Using https://huw-wrangler2-559wwx4hvq6x.github.dev/.

**Reply by jungaretti** [reply]
Thanks for the followup. We are looking into this!

**Reply by jungaretti** [reply]
@huw We've deployed a fix for this bug. Could you try again and let me know how it goes?

**Reply by huw** [reply]
Fixed! Thank you so much ❤️
